yeonjae/baekjoon/BFS_DFS/2667_단지번호붙이기.js

```diff
@@ -0,0 +1,67 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  const n = Number(input.shift());
+  const map = input.map((v) => v.split("").map(Number));
+  const answer = [];
+
+  const moves = [
+    [-1, 0],
+    [1, 0],
+    [0, 1],
+    [0, -1],
+  ];
+
+  const getIsValid = (ny, nx) => {
+    if (ny >= 0 && ny < n && nx >= 0 && nx < n && map[ny][nx]) {
+      return true;
+    }
+
+    return false;
+  };
+
+  const bfs = (start) => {
+    const queue = [start];
+    let count = 0;
+
+    while (queue.length) {
+      const [curY, curX] = queue.shift();
+      count++;
+
+      for (let i = 0; i < 4; i++) {
+        const ny = curY + moves[i][1];
+        const nx = curX + moves[i][0];
+
+        if (getIsValid(ny, nx)) {
+          map[ny][nx] = 0;
+          queue.push([ny, nx]);
+        }
+      }
```

Comment:
```suggestion
movers.forEach(move => {
  const [dy, dx] = move;
  const [ny, nx] = [curY + dy, curX + dx];
  // ...
})
```
- moves를 forEach를 활용한 함수형으로 짜면 좀 더 가독성 좋은 코드 작성이 가능합니다!

Reply:
오 이부분 반복문은 forEach로 쓰는게 더 가독성이좋네요! 코멘트 감사합니다!!